astronomer/providers/snowflake/extractors/snowflake_async_extractor.py

```diff
@@ -0,0 +1,162 @@
+
+        # (1) Parse sql statement to obtain input / output tables.
+        stm = "Sending SQL to parser {0}".format(self.operator.sql)
+        self.log.debug(stm)
+        sql_meta: Optional[SqlMeta] = parse(self.operator.sql, self.default_schema)
+        metadata = "Got meta {0}".format(sql_meta)
+        self.log.debug(metadata)
+
+        if not sql_meta:
+            return TaskMetadata(
+                name=task_name, inputs=[], outputs=[], run_facets=run_facets, job_facets=job_facets
+            )
+
+        # (2) Get Airflow connection
+        self.conn = get_connection(self._conn_id())
+
+        # (3) Default all inputs / outputs to current connection.
+        # NOTE: We'll want to look into adding support for the `database`
+        # property that is used to override the one defined in the connection.
+        source = Source(
+            scheme="snowflake", authority=self._get_authority(), connection_url=self._get_connection_uri()
+        )
+
+        database = self.operator.database
+        if not database:
+            database = self._get_database()
+
+        # (4) Map input / output tables to dataset objects with source set
+        # as the current connection. We need to also fetch the schema for the
+        # input tables to format the dataset name as:
+        # {schema_name}.{table_name}
+        inputs, outputs = get_table_schemas(
+            self._get_hook(),
+            source,
+            database,
+            self._information_schema_query(sql_meta.in_tables) if sql_meta.in_tables else None,
+            self._information_schema_query(sql_meta.out_tables) if sql_meta.out_tables else None,
+        )
+
+        query_ids = self._get_query_ids()
+        if len(query_ids) == 1:
+            run_facets["externalQuery"] = ExternalQueryRunFacet(
+                externalQueryId=query_ids[0], source=source.name
+            )
+        elif len(query_ids) > 1:
+            warnings_msg = (
+                "Found more than one query id for task {0}: {1} This might indicate that this task "
+                "might be better as multiple jobs".format(task_name, query_ids)
+            )
+            self.log.warning(warnings_msg)
+
+        return TaskMetadata(
+            name=task_name,
+            inputs=[ds.to_openlineage_dataset() for ds in inputs],
+            outputs=[ds.to_openlineage_dataset() for ds in outputs],
+            run_facets=run_facets,
+            job_facets=job_facets,
+        )
+
+    def _information_schema_query(self, tables: List[DbTableMeta]) -> str:
+        """
+        Forms the information execution query with table names and Returns SQL query
+
+        :param tables: List of table names
+        """
+        table_names = ",".join((f"'{self._normalize_identifiers(name.name)}'" for name in tables))
+        database = self.operator.database
+        if not database:
+            database = self._get_database()
+        # fmt: off
```

Comment:
Not sure actually we need `# fmt: off` or not can please try removing it

Reply:
I need this Because the this code was copied from the `openlineage` code we are using latest version of `bandit` and in opelineage `bandit` version is different If try to fix the query its failing in other python version